components/size_aware_image.jsx

```diff
@@ -110,18 +115,14 @@ export default class SizeAwareImage extends React.PureComponent {
         return width < MIN_IMAGE_SIZE || height < MIN_IMAGE_SIZE;
     }
 
-    handleLoad = (event) => {
+    handleLoad = () => {
         if (this.mounted) {
-            const image = event.target;
-            const isSmallImage = this.isSmallImage(image.naturalWidth, image.naturalHeight);
             this.setState({
                 loaded: true,
                 error: false,
-                isSmallImage,
-                imageWidth: image.naturalWidth,
             }, () => { // Call onImageLoaded prop only after state has already been set
-                if (this.props.onImageLoaded && image.naturalHeight) {
-                    this.props.onImageLoaded({height: image.naturalHeight, width: image.naturalWidth});
+                if (this.props.onImageLoaded && this.state.imageHeight) {
+                    this.props.onImageLoaded({height: this.state.imageHeight, width: this.state.imageWidth});
```

Comment:
Similarly, this should probably pass the dimensions like `image.naturalWidth || this.state.imageWidth` so that it uses the actual size of the image if that's available.